Bug fix for test_range and test_zoom_level

hanyelgaml-roche opened the pull request.

jkanche committed:
cleaning up a few things

jkanche committed:
cleanup req

jkanche committed:
add github actions

jkanche committed:
cleanup msgpack

jkanche committed:
more lint errors, update actions

jkanche committed:
add gh badge

jkanche committed:
update readme

jkanche committed:
update config

jkanche committed:
update docs

jkanche committed:
add links to rtd

jkanche committed:
update docs

jkanche committed:
update readme

Commit:
Final commit before updating the fork

hanyelgaml-roche committed:
Merge pull request #1 from epiviz/main

update from upstream

Commit:
Merge branch 'main' of https://github.com/hanyelgaml-roche/efs_parser into main

Commit:
removing bug

Commit:
resolving the BigBed returned range bug.

Commit:
resolving the BigBed returned range bug.

Commit:
resolving the BigBed returned range bug.

Commit:
resolving the BigBed returned range bug.

Commit:
bug fix and test_range test_zoom_level

jkanche commented on src/efs_parser/BigWig.py:
```diff
@@ -184,7 +184,15 @@ def getRange(self, chr, start, end, bins=2000, zoomlvl=-1, metric="AVG", respTyp
                 result["chr"] = chr
+                # checking range by: Hany Elgaml
 
+                query_string = '(start >= ' + str(start) + ' and start <= ' + str(end) + ') or (end >= ' + str(
+                    start) + ' and end <= ' + str(end) + ')'
```
we can simplify this ever further 

```
query_string = f"start <= {end} and end >= {start}" 
```

also checkout f strings in python

hanyelgaml-roche replied:
Done

jkanche commented on src/efs_parser/BigWig.py:
```diff
@@ -184,7 +184,15 @@ def getRange(self, chr, start, end, bins=2000, zoomlvl=-1, metric="AVG", respTyp
+                query_string = '(start >= ' + str(start) + ' and start <= ' + str(end) + ') or (end >= ' + str(
+                    start) + ' and end <= ' + str(end) + ')'
+                result = result.query(query_string)
+                result.insert(1, "new", range(0, len(result)))
+                result.set_index('new', inplace=True)
```
we don't need to reindex the result, so lets remove these lines

hanyelgaml-roche replied:
done

jkanche commented on tests/test_BigBed_remote.py:
```diff
@@ -31,8 +31,8 @@ def test_range():
-    assert(len(res) == 437)
-    for i in range(0,437):
+    assert(len(res) == 102)
+    for i in range(0,102):
```
pandas provides a method `iterrows` to iterate through the dataframe. That might be more appropriate here. 

I would also avoid checking explicit lengths

jkanche replied:
the `test_range` for BigWigs also need to check if the result is within the range we query

hanyelgaml-roche replied:
done

jkanche commented:
Also, your commit includes your entire `venv` directory. Please make sure to remove this in another commit

jkanche commented:
Looking at why the build is failing, we have a couple of more issues - 

For tests with `TranscriptTbx` and `GTFParsed`, use the following urls instead of file paths - 

- transcripts - https://raw.github.com/epiviz/efs-genomes/master/hg38/transcripts.tsv.bgz
- genes - https://raw.github.com/epiviz/efs-genomes/master/hg38/genes.tsv.gz

jkanche commented:
there's duplication of commits in the PR.

I suggest re-forking the repository and making these changes again (without the `venv` folder), 
you can either hard reset your `main` to `epiviz/main` or pull in `epiviz/main` as a new branch into your local repo. either way should work. 

a few links on git and github

- https://guides.github.com/
- https://medium.com/automationschool/git-super-basic-primer-for-beginner-developers-and-qa-part-1-b62e46863f75
- https://www.atlassian.com/git

Commit:
simplifying query string to be:
query_string = f"start <= {end} and end >= {start}"
- removing re-indexing from the filtered dataframe
- the use of iterrows for looping
removing explicit lengthes in test_range() in both:
- test_BigBed_remote.py
- test_BigWig_remote.py

For tests with TranscriptTbx , use the following urls instead of file paths -

transcripts - https://raw.github.com/epiviz/efs-genomes/master/hg38/transcripts.tsv.bgz

Commit:
simplifying query string to be:
query_string = f"start <= {end} and end >= {start}"
- removing re-indexing from the filtered dataframe
- the use of iterrows for looping
removing explicit lengthes in test_range() in both:
- test_BigBed_remote.py
- test_BigWig_remote.py

For tests with TranscriptTbx , use the following urls instead of file paths -

transcripts - https://raw.github.com/epiviz/efs-genomes/master/hg38/transcripts.tsv.bgz

Commit:
new fork commit

Commit:
removing venv directory
